src/utils.jl

```diff
@@ -44,19 +44,83 @@ vcat_arrays_vector(arrays_vector::AbstractVector{Param}) = vcat(Knet.cat1d.(arra
 """
     reset_minibatch_train!(nlp::AbstractKnetNLPModel)
 
-Select a new training minibatch for `nlp`.
-Typically used before a new evaluation of the loss function/gradient.
+Select the first training minibatch for `nlp`.
 """
-reset_minibatch_train!(nlp::AbstractKnetNLPModel) =
-  nlp.current_training_minibatch = rand(nlp.training_minibatch_iterator)
+function reset_minibatch_train!(nlp::AbstractKnetNLPModel)
+  nlp.current_training_minibatch = first(nlp.training_minibatch_iterator)
+  nlp.i_train = 1
+end
+
+"""
+  rand_minibatch_train!(nlp::AbstractKnetNLPModel)
+
+Select a training minibatch for `nlp` randomly.
+"""
+function rand_minibatch_train!(nlp::AbstractKnetNLPModel)
+  nlp.i_train = rand(1:(nlp.training_minibatch_iterator.imax))
+  nlp.current_training_minibatch = iterate(nlp.training_minibatch_iterator, nlp.i_train)
```

Comment:
What is the point of this method? Why not just grab the next mini batch in the iterator?

Reply:
This is for selecting a random batch, 
as some of the DNN training, instead of all data, they select a minibatch as random, this way we support that as well

Reply:
The PR goal is to provide several choices to select a training mini-batch from the training dataset.
This method is the current `reset_minibatch_train!` from main renamed, we didn't want to make KnetNLPModel functionnalities regress.
We changed its name for consistency.